src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,250 @@
+                PreparedStatement statement
+                        = connection.prepareStatement(updateRequest)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in cars DB.", e);
+        }
+        deletePreviousRelations(car.getId());
+        setCarToDriversRelation(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        deletePreviousRelations(id);
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Long> carsId = new ArrayList<>();
+        String getAllByDriverRequest =
+                "SELECT cars_id "
+                        + "FROM cars_drivers cd "
+                        + "JOIN cars c ON cd.cars_id = c.id "
+                        + "WHERE drivers_id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(getAllByDriverRequest)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carsId.add(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+        List<Car> cars = new ArrayList<>();
+        extractCarsByCarsId(carsId, cars);
+        return cars;
+    }
+
+    private void extractCarsByCarsId(List<Long> carsId, List<Car> cars) {
+        if (!carsId.isEmpty()) {
+            for (Long carId : carsId) {
+                cars.add(get(carId).orElseThrow(() ->
+                        new DataProcessingException("Can't get car by id: "
+                                + carId + " from cars DB.")));
+            }
+        }
+    }
+
+    private void deletePreviousRelations(Long id) {
+        Car previousCarVersion = this.get(id).orElseThrow(() ->
+                new DataProcessingException("Can't get existing variant of car with id: "
+                        + id + " from cars DB."));
+        if (previousCarVersion.getDrivers().isEmpty()) {
+            return;
+        }
+        String insertRequest = "DELETE FROM cars_drivers "
```

Comment:
```suggestion
        String insertRequest = "DELETE FROM cars_drivers "
```
Incorrect variable name. Please fix the same issue in all places. Also let's use 'query' suffix instead of 'request' as request is usually used for http requests